cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/StreamToDataset.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.examples.wikipedia;
+
+import co.cask.cdap.api.Resources;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.stream.StreamBatchReadable;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import co.cask.cdap.api.mapreduce.AbstractMapReduce;
+import co.cask.cdap.api.mapreduce.MapReduceContext;
+import co.cask.cdap.api.workflow.Value;
+import co.cask.cdap.api.workflow.WorkflowToken;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * MapReduce program to dump a list of likes to a dataset
+ */
+public class StreamToDataset extends AbstractMapReduce {
+  private static final Logger LOG = LoggerFactory.getLogger(StreamToDataset.class);
+
+  private final String name;
+
+  public StreamToDataset(String name) {
+    this.name = name;
+  }
+
+  @Override
+  public void configure() {
+    setName(name);
+    setDescription("A MapReduce program that dumps page titles from a user's likes on facebook to a dataset.");
+    setMapperResources(new Resources(512));
+  }
+
+  @Override
+  public void beforeSubmit(MapReduceContext context) throws Exception {
+    Job job = context.getHadoopJob();
+    job.setNumReduceTasks(0);
+    WorkflowToken workflowToken = context.getWorkflowToken();
+    Class<? extends Mapper> mapper = PageTitleToDatasetMapper.class;
+    String inputStream = WikipediaPipelineApp.PAGE_TITLES_STREAM;
+    String outputDataset = WikipediaPipelineApp.PAGE_TITLES_DATASET;
+    if (workflowToken != null) {
+      Value likesToDatasetResult = workflowToken.get("result", WikipediaPipelineApp.LIKES_TO_DATASET_MR_NAME);
+      if (likesToDatasetResult != null && likesToDatasetResult.getAsBoolean()) {
+        // likes stream to dataset has already run and been successful in this run so far.
```

Comment:
The "likes" stream to the dataset has already run and has been successful in this run so far.
Now run raw wikipedia stream to dataset.